Description: Applications could not get the PPD file for
 statically-configured IPP-shared print queues
Author: Till Kamppeter <till.kamppeter@gmail.com>
Bug-Upstream: http://cups.org/str.php?L4178
Last-Updated: 2012-09-19

--- a/cups/util.c
+++ b/cups/util.c
@@ -1731,6 +1731,22 @@
 
       return (1);
     }
+    else if (device_uri &&
+	     (!strncmp(device_uri, "ipp:", 4) != NULL ||
+	      !strncmp(device_uri, "ipps:", 5) != NULL))
+    {
+     /*
+      * Statically-configured IPP shared printer.
+      */
+
+      httpSeparateURI(HTTP_URI_CODING_ALL,
+                      device_uri,
+                      scheme, sizeof(scheme), username, sizeof(username),
+		      host, hostsize, port, resource, resourcesize);
+      ippDelete(response);
+
+      return (1);
+    }
     else if ((attr = ippFindAttribute(response, "member-uris",
                                       IPP_TAG_URI)) != NULL)
     {
